### PR TITLE
Subscribe stop crash

### LINKED ISSUE #13028: Exception in onStop callback: TypeError: handle.stop is not a function
With Meteor 3 beta 4 and the small example app (the react skeleton app in meteor), I get this error when the web page is closed and the subscription is stopped.

```
I20240225-16:05:53.868(1)? Exception in onStop callback: TypeError: handle.stop is not a function
I20240225-16:05:53.878(1)?     at OplogObserveDriver._stop (packages/mongo/oplog_observe_driver.js:966:20)
I20240225-16:05:53.878(1)?     at OplogObserveDriver.stop (packages/mongo/oplog_observe_driver.js:971:12)
I20240225-16:05:53.878(1)?     at ObserveMultiplexer._stop (packages/mongo/observe_multiplex.js:85:5)
I20240225-16:05:53.878(1)?     at ObserveMultiplexer.removeHandle (packages/mongo/observe_multiplex.js:72:7)
I20240225-16:05:53.878(1)?     at ObserveHandle.stop (packages/mongo/observe_multiplex.js:230:5)
I20240225-16:05:53.878(1)?     at Subscription.<anonymous> (packages/mongo/collection.js:590:14)
I20240225-16:05:53.878(1)?     at Subscription.<anonymous> (packages/meteor.js:1400:19)
```

Repro:
Clone https://github.com/Meteor-Community-Packages/meteor-typescript/tree/meteor3/tests/small-typescript-app (the meteor3 branch) and run `meteor run`. Open localhost:3000 and then close the browser tab to see the error.

macos sonoma 14.3.1, MBP M1

### PR DESCRIPTION
**Pull request by perbergland**
Fixes #13028 and adds validation to catch more errors.

I suspect there are plenty more missing awaits, but at least these a few are fixed here.
Also turned var's into const's because it needs to be done some day.

I tried to preserve the sequence of calls between the constructor and the async _init method

### CODE REVIEW

**Comment by denihs** [comment]
Hi, @perbergland. Some tests are failing, but maybe it is just a matter of rerunning them.

Could you merge the release-3.0 branch here again? 

Let's see if the tests are going to pass.

I tried to merge release-3.0 here, but I cannot commit to your fork.

**Review by denihs — CHANGES_REQUESTED** [review]

File: packages/mongo/polling_observe_driver.js [hunk 1]
```diff
@@ -37,47 +37,49 @@ PollingObserveDriver = function (options) {
...
-      Meteor.clearInterval(intervalHandle);
-    });
-  }
+  
 };
 
 _.extend(PollingObserveDriver.prototype, {
   _init: async function () {
+    const listenersHandle = await listenAll(
+      self._cursorDescription, function (notification) {
```
[the remark this answers is not in the document]
Comment (perbergland) [reply packages/mongo/polling_observe_driver.js:46]: Thx will fix

[the remark this answers is not in the document]
Comment (perbergland) [reply packages/mongo/polling_observe_driver.js:49]: I can’t see what tests broke on circle-ci? Can I run them locally? In any case, the code is probably fixed now

**Review by Grubba27 — APPROVED** [review]
LGTM.

The failing test in the CI is expected(docs issue).

File: packages/mongo/polling_observe_driver.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (denihs) [reply packages/mongo/polling_observe_driver.js:49]: Yeap, to run locally, you can execute ` ./meteor test-packages `. But they're all passing now and look good to me! Great job!

**Review by denihs — APPROVED** [review]

**Pull request merged**